diff --git a/s3270/Makefile.obj.in b/s3270/Makefile.obj.in
index 0fe01cd..dfa563b 100644
--- a/s3270/Makefile.obj.in
+++ b/s3270/Makefile.obj.in
@@ -73,7 +73,7 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(BUILDCC) $(XCPPFLAGS) -o $@ $(filter %.c,$^)
+	$(BUILDCC) $(XCPPFLAGS) -o $@ $(filter %.c,$^)
 
 s3270: $(OBJS1) $(DEP3270) $(DEP32XX) $(DEP3270STUBS)
 	$(CC) -o $@ $(OBJS1) $(LDFLAGS) $(LD3270) $(LD32XX) $(LD3270STUBS) $(LIBS)
