diff --git a/include/glue.h b/include/glue.h
index f05f33c..e639eeb 100644
--- a/include/glue.h
+++ b/include/glue.h
@@ -43,6 +43,7 @@ void parse_xrm(const char *arg, const char *where);
 char *safe_string(const char *s);
 bool process_events(bool block);
 void cmdline_help(bool as_action);
+void elf_init(void);
 void model_init(void);
 typedef bool merge_profile_t(void);
 void register_merge_profile(merge_profile_t *);
