diff --git a/include/resources.h b/include/resources.h
index 03f08c3..5918965 100644
--- a/include/resources.h
+++ b/include/resources.h
@@ -97,6 +97,7 @@
 #define ResDisconnectClear	"disconnectClear"
 #define ResDoConfirms		"doConfirms"
 #define ResDpi			"dpi"
+#define ResElf      "elf"
 #define ResEmulatorFont		"emulatorFont"
 #define ResEof			"eof"
 #define ResErase		"erase"
@@ -532,6 +533,7 @@
 #define OptDefaultFgBg		"-defaultfgbg"
 #define OptDefScreen		"-defscreen"
 #define OptDevName		"-devname"
+#define OptElf        "-elf"
 #define OptEmulatorFont		"-efont"
 #define OptHostsFile		"-hostsfile"
 #define OptHelp1		"--help"
