diff --git a/Common/fb-c3270 b/Common/fb-c3270
index 52f98a8..a094e07 100644
--- a/Common/fb-c3270
+++ b/Common/fb-c3270
@@ -32,6 +32,8 @@ x3270.message.buildEnabled:			enabled
 x3270.message.buildOpts:			Build options:
 x3270.message.hostCodePage:			Host code page:
 x3270.message.unnegotiated:			TELNET negotiation incomplete
+x3270.expressLogon: false
+x3270.ibmApplId:
 #ifdef _WIN32
 x3270.message.windowsCodePage:			Windows code page:
 #endif
