diff --git a/Common/mkfb.c b/Common/mkfb.c
index 3b2f047..dca201b 100644
--- a/Common/mkfb.c
+++ b/Common/mkfb.c
@@ -103,23 +103,24 @@ usage(void)
 }
 
 /*
-* Wrapper around Windows' brain-dead tmpfile().
+* Wrapper around Windows' and z/OS' brain-dead tmpfile().
 */
 static FILE *
 mkfb_tmpfile(void)
 {
     FILE *f;
-#if defined(_WIN32) /*[*/
+#if defined(_WIN32) || defined(__MVS__) /*[*/
     char *n;
 #endif /*]*/
 
-#if !defined(_WIN32) /*[*/
+#if !defined(_WIN32) && !defined(__MVS__) /*[*/
     f = tmpfile();
     if (f == NULL) {
 	perror("tmpfile");
 	exit(1);
     }
 #else /*][*/
+#if defined(_WIN32)
     n = _tempnam(NULL, "mkfb");
     if (n == NULL) {
 	fprintf(stderr, "_tempnam failed.\n");
@@ -131,6 +132,19 @@ mkfb_tmpfile(void)
 		strerror(errno));
 	exit(1);
     }
+#else
+    n = tempnam(NULL, "mkfb");
+    if (n == NULL) {
+	fprintf(stderr, "tempnam failed.\n");
+	exit(1);
+    }
+    f = fopen(n, "w+");
+    if (f == NULL) {
+	fprintf(stderr, "tempnam open(\"%s\") failed: %s\n", n,
+		strerror(errno));
+	exit(1);
+    }
+#endif
     free(n);
 #endif /*]*/
 
