diff --git a/Common/mkfb.c b/Common/mkfb.c
index 3b2f047..98a5c68 100644
--- a/Common/mkfb.c
+++ b/Common/mkfb.c
@@ -103,17 +103,17 @@ usage(void)
 }
 
 /*
-* Wrapper around Windows' brain-dead tmpfile().
+* Wrapper around Windows' and z/OS' brain-dead tmpfile().
 */
 static FILE *
 mkfb_tmpfile(void)
 {
     FILE *f;
-#if defined(_WIN32) /*[*/
+#if defined(_WIN32) || defined(__MVS__) /*[*/
     char *n;
 #endif /*]*/
 
-#if !defined(_WIN32) /*[*/
+#if !defined(_WIN32) && !defined(__MVS__) /*[*/
     f = tmpfile();
     if (f == NULL) {
 	perror("tmpfile");
@@ -125,7 +125,11 @@ mkfb_tmpfile(void)
 	fprintf(stderr, "_tempnam failed.\n");
 	exit(1);
     }
+#if defined(_WIN32)
     f = fopen(n, "w+b");
+#else
+    f = fopen(n, "w+");
+#endif
     if (f == NULL) {
 	fprintf(stderr, "_tempnam open(\"%s\") failed: %s\n", n,
 		strerror(errno));
