--- telnet_new_environ.c.orig	2026-04-02 20:53:59
+++ telnet_new_environ.c	2026-04-03 19:53:33
@@ -553,32 +553,30 @@
 
 	    /* Dump one entry. */
 	    value = find_environ(
-		    (ereq->group == TELOBJ_VAR)? &vars : &uservars, ereq->name,
-		    ereq->name_len);
+	     (ereq->group == TELOBJ_VAR)? &vars : &uservars, ereq->name,
+	     ereq->name_len);
 
-	    vb_appendf(&reply, "%c", ereq->group);
-	    vb_append(&reply, ereq->name, ereq->name_len);
+	    /* Only respond if we have a value for this variable (match PCOMM behavior) */
 	    if (value != NULL) {
-		vb_appendf(&reply, "%c", TELOBJ_VALUE);
-		if (value->devname != NULL) {
-		    dnext = devname_next(value->devname);
+	 vb_appendf(&reply, "%c", ereq->group);
+	 vb_append(&reply, ereq->name, ereq->name_len);
+	 vb_appendf(&reply, "%c", TELOBJ_VALUE);
+	 if (value->devname != NULL) {
+	     dnext = devname_next(value->devname);
 
-		    vb_append(&reply, dnext, strlen(dnext));
-		} else {
-		    vb_append(&reply, value->value, value->value_len);
-		}
-	    }
+	     vb_append(&reply, dnext, strlen(dnext));
+	 } else {
+	     vb_append(&reply, value->value, value->value_len);
+	 }
 
-	    /* Trace the reply, */
-	    vb_appendf(&trace_out, " %s \"%s\"",
-		    telobjs[ereq->group],
-		    expand_name(ereq->name, ereq->name_len));
-	    if (value != NULL) {
-		vb_appendf(&trace_out, " %s \"%s\"",
-		    telobjs[TELOBJ_VALUE],
-		    (dnext != NULL)?
-			expand_name(dnext, strlen(dnext)):
-			expand_name(value->value, value->value_len));
+	 /* Trace the reply */
+	 vb_appendf(&trace_out, " %s \"%s\" %s \"%s\"",
+	  telobjs[ereq->group],
+	  expand_name(ereq->name, ereq->name_len),
+	  telobjs[TELOBJ_VALUE],
+	  (dnext != NULL)?
+	      expand_name(dnext, strlen(dnext)):
+	      expand_name(value->value, value->value_len));
 	    }
 	}
     } FOREACH_LLIST_END(ereqs, ereq, ereq_t);
