diff --git a/c3270/Test/testIbmHosts.py b/c3270/Test/testIbmHosts.py
index 86eec2f..598a796 100755
--- a/c3270/Test/testIbmHosts.py
+++ b/c3270/Test/testIbmHosts.py
@@ -40,6 +40,7 @@ import Common.Test.playback as playback
 import Common.Test.cti as cti
 
 @unittest.skipIf(sys.platform.startswith('win'), "Windows uses different c3270 graphic tests")
+@unittest.skipIf(sys.platform == 'zos', "z/OS does not close socket")
 class TestC3270IbmHosts(cti.cti):
 
     # Drain the PTY.
