diff --git a/c3270/Makefile.obj.in b/c3270/Makefile.obj.in
index e640ffb..b1a2d16 100644
--- a/c3270/Makefile.obj.in
+++ b/c3270/Makefile.obj.in
@@ -75,7 +75,10 @@ fallbacks.c: mkfb $(FALLBACKS)
 mkfb: mkfb.c
 	$(BUILDCC) $(XCPPFLAGS) -o $@ $(filter %.c,$^)
 
-mkkeypad: mkkeypad.c
+mkkeypad: mkkeypad.o
+	$(CC) -o $@ $< $(CFLAGS) $(LDFLAGS) @LIBS@
+
+mkkeypad.o: mkkeypad.c
 	$(BUILDCC) $(XCPPFLAGS) -o $@ $<
 
 keypad.o: compiled_keypad.h
