diff --git a/c3270/Makefile.obj.in b/c3270/Makefile.obj.in
index e640ffb..95d369c 100644
--- a/c3270/Makefile.obj.in
+++ b/c3270/Makefile.obj.in
@@ -73,13 +73,16 @@ fallbacks.c: mkfb $(FALLBACKS)
 	./mkfb -c -o $@ $(filter-out mkfb,$^)
 
 mkfb: mkfb.c
-	$(BUILDCC) $(XCPPFLAGS) -o $@ $(filter %.c,$^)
+	$(BUILDCC) $(XCPPFLAGS) -o $@ $(filter %.c,$^) $(LDFLAGS) $LIBS
 
 mkkeypad: mkkeypad.c
 	$(BUILDCC) $(XCPPFLAGS) -o $@ $<
 
 keypad.o: compiled_keypad.h
 
+mkkeypad: mkkeypad.o
+	$(CC) -o $@ $< $(CFLAGS) $(LDFLAGS) $LIBS
+
 compiled_keypad.h: keypad.full keypad.labels keypad.map keypad.outline keypad.callbacks mkkeypad
 	./mkkeypad -I$(TOP)/Common/c3270 >$@
 
