diff --git a/c3270/Makefile.obj.in b/c3270/Makefile.obj.in
index e640ffb..f1dcfbb 100644
--- a/c3270/Makefile.obj.in
+++ b/c3270/Makefile.obj.in
@@ -80,6 +80,9 @@ mkkeypad: mkkeypad.c
 
 keypad.o: compiled_keypad.h
 
+mkkeypad: mkkeypad.o
+	$(CC) -o $@ $< $(CFLAGS) $(LDFLAGS) @LIBS@
+
 compiled_keypad.h: keypad.full keypad.labels keypad.map keypad.outline keypad.callbacks mkkeypad
 	./mkkeypad -I$(TOP)/Common/c3270 >$@
 
