diff --git a/include/appres.h b/include/appres.h
index 5479bf6..0b2cace 100644
--- a/include/appres.h
+++ b/include/appres.h
@@ -57,6 +57,7 @@ typedef struct {
     bool	 debug_tracing;
     char	*devname;	/* for 5250 */
     bool	 disconnect_clear;
+    bool	 express_logon;
     bool	 extended_data_stream;
     char	*ft_command;
 #if defined(_WIN32) /*[*/
@@ -66,6 +67,7 @@ typedef struct {
     char	*hostname;
     char	*hostsfile;
     char	*httpd_port;
+    char        *ibm_applid;
     char	*idle_command;
     bool	 idle_command_enabled;
     char	*idle_timeout;
