diff --git a/Common/Test/json_test.c b/Common/Test/json_test.c
index 00edd48..7066c21 100644
--- a/Common/Test/json_test.c
+++ b/Common/Test/json_test.c
@@ -116,7 +116,7 @@ main(int argc, char *argv[])
 
     /* Get ready for stderr redirection when we expect an assertion to fail. */
 #if !defined(_WIN32) /*[*/
-    dev_null = open("/dev/null", 0);
+    dev_null = open("/dev/null", O_WRONLY);
 #else /*][*/
     dev_null = open("nul", 0);
 #endif /*]*/
